### ISSUE TITLE
feature: settings export/import

### ISSUE THREAD

**Issue by bughit** [piroor/treestyletab#2190]
1. If a bug does not occur with default settings, it should be possible to export and attach the settings that produce it. There may have been a lot of settings tweaked and one should not have to search through them all to report a bug. So there are 2 related scenarios here:
   1. exporting settings from a main profile to a new profile
   2. exporting settings to include in a bug report
3. Sometimes to identify a regressions (and possibly revert to an older version) one has to uninstall a current one.  When I did that all settings were reset.

**Comment by piroor** [comment]
TST already has the feature: TST options => Development => Debug mode => All Configs => look at bottom of the table please.

**Comment by bughit** [comment]
thanks, a few comments

1. It's too hidden, should be more discoverable. Activating "debug mode" is typically associated with extra activity that slows down the app, so one is not inclined to do that without a good reason, and there is no intuitive expectation that doing so would reveal settings export ui.
2. Does activating debug mode in TST do something beyond showing additional ui?

**Comment by piroor** [comment]
Sorry but basically I'm still negative to support import/export all configurations of TST.

* TST is designed to synchronize most options via Firefox Sync, and it is the recommended method to share configs between multiple devices. TST has some platform-specific options, and they should not be shared between devices. TST ignores such environment-specific options on the Firefox Sync, but doesn't on import/export.
* Import/export should be done only on the initial time on the environment, and too frequent import/export is not an expected situation. If you need to do that, it should be resolved fundamentally with different solution - more intelligent detection of suitable behavior on various situations, a remote profile, Firefox Sync with a local server, or others.

In other words, I don't want the import/export feature to be used widely by regular people. Exposing the feature is in the opposite of my design policy.

**Comment by vonHabsi** [comment]
Does the import/export setting include keyboard shortcuts?

**Comment by piroor** [comment]
The feature doesn't export user-defined keyboard shortcuts, because they are stored in different datastore.

**Comment by irvinm** [comment]
@piroor given this is contrary to how you want the addon to be used and will not make the change, I recommend we close this issue.

Is there any need to include the "Debug mode => All Configs" information as part of the "contributing guidelines" for those who should include configuration information?

**Comment by piroor** [comment]
@irvinm Thank you for suggestion, I've updated the contribution guide.

**Comment by irvinm** [comment]
@piroor any reason to leave this open now?

**Comment by piroor** [comment]
I missed this. Today "Import" and "Export" buttons are placed under the "Development" section directly, so I think this has already fixed.
